tests/test_generate_reports.py

```diff
@@ -0,0 +1,357 @@
+                "key": group_number,
+                "amount": transaction["amount"],
+                "references": [transaction["referenceText"]],
+                "groupName": transaction["groupName"],
+            }
+        else:
+            grouped["amount"] += transaction["amount"]
+            grouped["references"] += [transaction["referenceText"]]
+
+    grouped_by_ref_list = list(grouped_by_ref.values())
+    return grouped_by_ref_list
+
+
+def _expected_categ_group(
+    grouped_transactions: List[GroupedTransaction],
+    group_references: List[set[str]],
+) -> List[Any]:
+    transactions = deepcopy(grouped_transactions)
+    transactions_ref = deepcopy(group_references)
+    categories_group: Dict[Any, Any] = {}
+
+    for transaction in transactions:
+        key_category = get_category(
+            transaction["referenceText"],
+            transactions_ref[transaction["groupNumber"]],
+            fallback_reference=transaction["groupName"],
+        )
+        category_group = categories_group.get(key_category)
+        if category_group is None:
+            categories_group[key_category] = {
+                "key": key_category,
+                "amount": transaction["amount"],
+                "references": [transaction["referenceText"]],
+            }
+        else:
+            category_group["amount"] += transaction["amount"]
+            category_group["references"] += [transaction["referenceText"]]
+
+    categories_group_list = list(categories_group.values())
+    return categories_group_list
+
+
+def _expected_categ_transactions(
+    grouped_transactions: List[GroupedTransaction],
+    group_references: List[set[str]],
+) -> List[Any]:
+
+    categorized_transactions: List[Any] = deepcopy(grouped_transactions)
+    categorized_transactions_ref = deepcopy(group_references)
+
+    for transaction in categorized_transactions:
+        key_category = get_category(
+            transaction["referenceText"],
+            categorized_transactions_ref[transaction["groupNumber"]],
+            fallback_reference=transaction["groupName"],
+        )
+        transaction["customCategory"] = key_category
+
+    return categorized_transactions
+
+
+def test_generate_reports_output(write_json_mock: Mock) -> None:
+
+    transactions_test_file_path = "tests/test_data/transactions/test_transaction.json"
+    configuration_test_file_path = "tests/test_data/config/test_config_file.yaml"
+    cache_user_configuration(configuration_test_file_path)
+
+    transactions_raw = _open_transaction_test_file(transactions_test_file_path)
+    incomes, expenses = _split_transactions(transactions_raw)
+    grouped_income, grouped_income_ref = group_transactions(
+        incomes, TransactionGroupingType.ReferenceSimilarity
+    )
+    grouped_expense, grouped_expense_ref = group_transactions(
+        expenses, TransactionGroupingType.ReferenceSimilarity
+    )
+
+    clear_user_configuration_cache()
```

Comment:
The whole block reproduces a real code business logic, which can be buggy, then it defeats the purpose of testing the real code.

I suggest removing this block and carry on the test with written/validated files for what should be the ideal output.

Reply:
this block has been removed